src/test/ui/assume-type-intrinsics.rs

```diff
@@ -0,0 +1,11 @@
+#![feature(never_type)]
+#![feature(const_maybe_uninit_assume_init)]
+
+fn main() {
+    use std::mem::MaybeUninit;
+
+    const _BAD: () = unsafe {
+        MaybeUninit::<!>::uninit().assume_init();
+        //~^ WARN: the type `!` does not permit being left uninitialized
```

Comment:
Please test for the error, not the warning. The warning has nothing to do with your PR, it is caused by the `invalid_value` lint.

Actually, please add `#[allow(invalid_value)]` to ensure that the lint doesn't have any effect here.

Reply:
Where should I put the error comment when the error complains about what happens *inside* the call to `assume_init`?

```
error: any use of this value will cause an error                                                                                       
  --> /home/albin/rust/rustc/library/core/src/mem/maybe_uninit.rs:514:13                                                               
   |                                                                                                                                   
LL |               intrinsics::assert_inhabited::<T>();                                                                                
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                 
   |               |                                                                                                                   
   |               attempted to instantiate uninhabited type `!`                                                                       
   |               inside `MaybeUninit::<!>::assume_init` at /home/albin/rust/rustc/library/core/src/mem/maybe_uninit.rs:514:13        
   |               inside `_BAD` at /home/albin/rust/rustc/src/test/ui/assume-type-intrinsics.rs:9:9                                   
   |                                                                                                                                   
  ::: /home/albin/rust/rustc/src/test/ui/assume-type-intrinsics.rs:8:5                                                                 
   |                                                                                                                                   
LL | /     const _BAD: () = unsafe {              
LL | |         MaybeUninit::<!>::uninit().assume_init();                                             
LL | |                                                             
LL | |     };                                     
   | |______-                                     
   |                                              
   = note: `#[deny(const_err)]` on by default     

error: aborting due to previous error
```

I have tried 
```rust
const _BAD: () = unsafe {
    MaybeUninit::<!>::uninit().assume_init();
    //~^ ERROR: any use of this value will cause an error
};
```

but that does not seem to be it:

```
error: /home/albin/rust/rustc/src/test/ui/assume-type-intrinsics.rs:9: expected error not found: any use of this value will cause an error
                                                                                                                                                                                   
error: 0 unexpected errors found, 1 expected errors not found
status: exit code: 1
command: [...]
not found errors (from test file): [
    Error {
        line_num: 9,
        kind: Some(
            Error,
        ),
        msg: "any use of this value will cause an error",
    },
]
```

Reply:
You need ` // error-pattern`: https://github.com/rust-lang/rust/pull/76143/files#diff-23706e664a2a576e6e031e0be43b04ff0f1916b93c2358b518d67e53aa92fbc3L1

I should probably add that to the dev-guide ...

Reply:
Thank you :)